JIT assert on arm64 when running DllImportGenerator tests: op2->TypeIs(TYP_LONG)

https://dev.azure.com/dnceng/public/_build/results?buildId=1429275&view=ms.vss-test-web.build-test-results-tab
```
/root/helix/work/correlation/dotnet exec --runtimeconfig DllImportGenerator.Tests.runtimeconfig.json --depsfile DllImportGenerator.Tests.deps.json xunit.console.dll DllImportGenerator.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing 
popd
===========================================================================================================
/root/helix/work/workitem/e /root/helix/work/workitem/e
  Discovering: DllImportGenerator.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  DllImportGenerator.Tests (found 94 test cases)
  Starting:    DllImportGenerator.Tests (parallel test collections = on, max threads = 4)

Assert failure(PID 27 [0x0000001b], Thread: 40 [0x0028]): Assertion failed 'op2->TypeIs(TYP_LONG)' in 'Arrays:AndAllMembers(SharedTypes.BoolStruct[],int):bool' during 'Lowering nodeinfo' (IL size 180)

    File: /__w/2/s/src/coreclr/jit/gentree.cpp Line: 2449
    Image: /root/helix/work/correlation/dotnet
```
also seen in some PGO testing I did.

This assertion failed the `CoreCLR Pri0 Runtime Tests Run Composite OSX arm64 checked` build: https://dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_apis/build/builds/1500310/logs/416:

```
  Building Targeting Pack
  external -> 
  Compiling framework using Crossgen2: "/Users/runner/work/1/s/dotnet.sh" "/Users/runner/work/1/s/artifacts/tests/coreclr/OSX.arm64.Checked/Tests/Core_Root/R2RTest/R2RTest.dll" compile-framework -cr "/Users/runner/work/1/s/artifacts/tests/coreclr/OSX.arm64.Checked/Tests/Core_Root" --output-directory "/Users/runner/work/1/s/artifacts/tests/coreclr/obj/OSX.arm64.Checked/crossgen.out" --release --nocleanup --target-arch arm64 -dop 4 -m "/Users/runner/work/1/s/artifacts/tests/coreclr/OSX.arm64.Checked/Tests/Core_Root/StandardOptimizationData.mibc" --composite --verify-type-and-field-layout --crossgen2-path "/Users/runner/work/1/s/artifacts/bin/coreclr/OSX.arm64.Checked/x64/crossgen2/crossgen2.dll"
  Using dotnet: /Users/runner/work/1/s/.dotnet/dotnet
  1 / 1 (0%, 0 failed): launching: /Users/runner/work/1/s/.dotnet/dotnet /Users/runner/work/1/s/artifacts/bin/coreclr/OSX.arm64.Checked/x64/crossgen2/crossgen2.dll @/Users/runner/work/1/s/artifacts/tests/coreclr/obj/OSX.arm64.Checked/crossgen.out/framework-r2r.dll.rsp
  1 / 1 (100%, 1 failed): failed in 149283 msecs, exit code 133, expected 0: dotnet /Users/runner/work/1/s/artifacts/bin/coreclr/OSX.arm64.Checked/x64/crossgen2/crossgen2.dll @/Users/runner/work/1/s/artifacts/tests/coreclr/obj/OSX.arm64.Checked/crossgen.out/framework-r2r.dll.rsp
    /Users/runner/work/1/s/src/coreclr/jit/gentree.cpp:2449
    Assertion failed 'op2->TypeIs(TYP_LONG)' in 'System.Collections.Generic.ArraySortHelper`1:InsertionSort(System.Span`1[Microsoft.Diagnostics.Tracing.CLRRuntimeActivityComputer+StartStopThreadEventData],System.Comparison`1[Microsoft.Diagnostics.Tracing.CLRRuntimeActivityComputer+StartStopThreadEventData])' during 'Lowering nodeinfo' (IL size 116)
/Users/runner/work/1/s/src/tests/build.proj(519,5): error MSB3073: The command ""/Users/runner/work/1/s/dotnet.sh" "/Users/runner/work/1/s/artifacts/tests/coreclr/OSX.arm64.Checked/Tests/Core_Root/R2RTest/R2RTest.dll" compile-framework -cr "/Users/runner/work/1/s/artifacts/tests/coreclr/OSX.arm64.Checked/Tests/Core_Root" --output-directory "/Users/runner/work/1/s/artifacts/tests/coreclr/obj/OSX.arm64.Checked/crossgen.out" --release --nocleanup --target-arch arm64 -dop 4 -m "/Users/runner/work/1/s/artifacts/tests/coreclr/OSX.arm64.Checked/Tests/Core_Root/StandardOptimizationData.mibc" --composite --verify-type-and-field-layout --crossgen2-path "/Users/runner/work/1/s/artifacts/bin/coreclr/OSX.arm64.Checked/x64/crossgen2/crossgen2.dll"" exited with code 1.
```

--- a/src/coreclr/jit/importer.cpp
+++ b/src/coreclr/jit/importer.cpp
@@ -4164,7 +4164,7 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
 
                 if (elemSize != 1)
                 {
-                    GenTree* sizeofNode = gtNewIconNode(elemSize);
+                    GenTree* sizeofNode = gtNewIconNode(static_cast<ssize_t>(elemSize), TYP_I_IMPL);
                     index               = gtNewOperNode(GT_MUL, TYP_I_IMPL, index, sizeofNode);
                 }
 

--- a/src/libraries/System.Runtime.InteropServices/tests/DllImportGenerator.Tests/ArrayTests.cs
+++ b/src/libraries/System.Runtime.InteropServices/tests/DllImportGenerator.Tests/ArrayTests.cs
@@ -301,7 +301,6 @@ namespace DllImportGenerator.IntegrationTests
         [Theory]
         [InlineData(true)]
         [InlineData(false)]
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/60624", typeof(PlatformDetection), nameof(PlatformDetection.IsNotMonoRuntime), nameof(PlatformDetection.IsArm64Process))]
         public void ArrayWithSimpleNonBlittableTypeMarshalling(bool result)
         {
             var boolValues = new[]

--- a/src/libraries/System.Runtime.InteropServices/tests/DllImportGenerator.Tests/CollectionTests.cs
+++ b/src/libraries/System.Runtime.InteropServices/tests/DllImportGenerator.Tests/CollectionTests.cs
@@ -195,7 +195,6 @@ namespace DllImportGenerator.IntegrationTests
         [Theory]
         [InlineData(true)]
         [InlineData(false)]
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/60624", typeof(PlatformDetection), nameof(PlatformDetection.IsNotMonoRuntime), nameof(PlatformDetection.IsArm64Process))]
         public void CollectionWithSimpleNonBlittableTypeMarshalling(bool result)
         {
             var boolValues = new List<BoolStruct>

--- a/src/libraries/System.Runtime.InteropServices/tests/DllImportGenerator.Tests/SpanTests.cs
+++ b/src/libraries/System.Runtime.InteropServices/tests/DllImportGenerator.Tests/SpanTests.cs
@@ -133,7 +133,6 @@ namespace DllImportGenerator.IntegrationTests
         [Theory]
         [InlineData(true)]
         [InlineData(false)]
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/60624", typeof(PlatformDetection), nameof(PlatformDetection.IsNotMonoRuntime), nameof(PlatformDetection.IsArm64Process))]
         public void SpanWithSimpleNonBlittableTypeMarshalling(bool result)
         {
             var boolValues = new BoolStruct[]

